spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/parameter/ParameterBindingParser.java

```diff
@@ -0,0 +1,295 @@
+package org.springframework.data.jdbc.repository.query.parameter;
```

Comment:
Spring Data contains a utility class from Commons (`org.springframework.data.repository.query.SpelQueryContext`) that allows analysis of SpEL queries and extraction of parameter bindings.

It would be used along the lines of:

```java
		List<ParameterBinding> parameterBindings = new ArrayList<>();

		SpelQueryContext queryContext = SpelQueryContext.of((counter, expression) -> {

			String parameterName = String.format("__synthetic_%d__", counter);
			parameterBindings.add(new ParameterBinding(parameterName, expression));
			return parameterName;
		}, String::concat);

		SpelQueryContext.SpelExtractor parsed = queryContext.parse(query);
```

which makes this class superfluous.